### ISSUE TITLE
how to generate schema.xml for Solr5?

### ISSUE THREAD

**Issue by skcc321** [sunspot/sunspot#711]
The problem is related to unknown field types.
After rake sunspot:solr:start i have got running solr 4.2.0 and configs for 4x version.
Solr5 service is installed using install script https://cwiki.apache.org/confluence/display/solr/Taking+Solr+to+Production

I can't create new core/collection using generated by sunspot scheme.xml/solrconfig.xml.
Generated scheme contains unknown field types. So what is the right way to generate solr5 supported scheme.xml?

**Comment by iRonin** [comment]
+1

**Comment by baldmountain** [comment]
+1

I installed my own version of solr (5.2.1 specifically) and managed to get it working by taking the generated scheme.xml file and merging it into the example scheme.xml file from the examplefiles directory of the solr-5.2.1 distribution. I also removed some of the generic field definitions from the example scheme.xml file. It seems to work, but I don't know if it is 100% correct. There are a few warnings about loading deprecated features in the solr log file.

**Comment by baldmountain** [comment]
@skcc321 @iRonin Try master rather than 2.2.0. I used the generated schema.xml and it seems to work fine with 5.2.1.

**Comment by satyagrahi** [comment]
+1

I just installed solr 5.3 (after upgrading linux to 14) and am struggling making it all work. Do you use jetty (we had jetty with Solr3 and used the .war files to configure)? And @baldmountain which example did you use (noschema?). Also, using java 7 or 8?

Am feeling that its better to use the packaged solr4 instance. Any idea when master is going to be released (2.3 or 3.0)?

**Comment by baldmountain** [comment]
@satyagrahi IIRC I changed my Gemfile to use master rather than 2.2.0. Like this:

# solr
gem 'sunspot_rails', :git => 'https://github.com/sunspot/sunspot.git'
group :development do
  gem 'sunspot_solr', :git => 'https://github.com/sunspot/sunspot.git'
end

Then I downloaded a version of my repository to the linux machine and ran it in dev mode so sunspot_solr would generate a schema.xml file. I then copied the auto generated schema.xml file to my solr config directory and it just worked with 5.2.1.

I think that was the process.

**Comment by satyagrahi** [comment]
Thanks @baldmountain !
First i removed java6, java7, and jetty6 which were around. 

Checked that the solr service was running:
```
$ sudo service solr status
Found 1 Solr nodes: 
Solr process 27985 running on port 8983
{
  "solr_home":"/var/solr/data/",
  "version":"5.3.0 1696229 - noble - 2015-08-17 17:10:43",
  "startTime":"2015-08-29T07:34:19.425Z",
  "uptime":"0 days, 8 hours, 19 minutes, 7 seconds",
  "memory":"84.5 MB (%17.2) of 490.7 MB"}
```

Then i created one core: 
`$ bin/solr create -c dev`

Then i followed your recommendation but got an error:
```
$ bundle exec rake sunspot:solr:reindex
rake aborted!
RSolr::Error::Http: RSolr::Error::Http - 400 Bad Request
Error: {'responseHeader'=>{'status'=>400,'QTime'=>2},'error'=>{'msg'=>'undefined field type','code'=>400}}

URI: http://localhost:8983/solr/dev/update?wt=ruby
Request Headers: {"Content-Type"=>"text/xml"}
Request Data: "<?xml version=\"1.0\" encoding=\"UTF-8\"?><delete><query>type:App</query></delete>"

Backtrace: /usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/rsolr-1.0.12/lib/rsolr/client.rb:284:in `adapt_response'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/rsolr-1.0.12/lib/rsolr/client.rb:190:in `execute'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/rsolr-1.0.12/lib/rsolr/client.rb:176:in `send_and_receive'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/bundler/gems/sunspot-c7be0bceb59e/sunspot_rails/lib/sunspot/rails/solr_instrumentation.rb:16:in `block in send_and_receive_with_as_instrumentation'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/activesupport-3.2.21/lib/active_support/notifications.rb:123:in `block in instrument'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/activesupport-3.2.21/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/activesupport-3.2.21/lib/active_support/notifications.rb:123:in `instrument'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/bundler/gems/sunspot-c7be0bceb59e/sunspot_rails/lib/sunspot/rails/solr_instrumentation.rb:15:in `send_and_receive_with_as_instrumentation'
(eval):2:in `post'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/rsolr-1.0.12/lib/rsolr/client.rb:82:in `update'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/rsolr-1.0.12/lib/rsolr/client.rb:146:in `delete_by_query'/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/rsolr-1.0.12/lib/rsolr/client.rb:284:in `adapt_response'
/usr/local/rvm/gems/ruby-1.9.3-p545@pwx/gems/rsolr-1.0.12/lib/rsolr/client.rb:190:in `execute'
```

I am not sure how to proceed. My sunspot config is:
```
dev:
  solr:
    hostname: localhost
    port: 8983
    path: /solr/dev
    log_level: WARNING
```

Thanks in advance!

ps: i would like to use the browser, but doesn't seem to work on my server (maybe port 8983 is blocked) and when i do curl, it just returns with nothing. How do i get to the solr admin interface remotely? `server_name:8983` or `server_name:8983/solr` get no answer or error.

**Comment by baldmountain** [comment]
try server_name:8983/solr/admin

Not sure about the error. Maybe master will work with solr 5.2 but not 5.3?

**Comment by herlon214** [comment]
+1

**Comment by UnConundrum** [comment]
I just spun up a new ubuntu server with tomcat8 and sold 5.2.1.  Unfortunately, my Rails app is still 2.3.11 (slowly upgrading to 4.2).  Since support for Rails 2x ended with Sunspot 2.0, is there any way I can get a working schema.xml?

**Comment by baldmountain** [comment]
@UnConundrum you might try building a new sample rails app with rails 4.2, sunspot master, and your db migrations and use the generated schema.xml from that sample app instead of the one in your app.

**Comment by UnConundrum** [comment]
I tried as you suggested and verified that sunspot 2.2 is installed.  I copied the schema.xml from solr/config to my server conf directory.  I still get an error:  Plugin init failure for [schema.xml] fieldType "date": Error loading class 'solr.DateField'. Schema file is /var/solr/data/production/conf/schema.xml

Somehow, I don't think this is the Solr 5 compatible schema.xml

**Comment by baldmountain** [comment]
Sunspot 2.2 isn't the right version. You need to use master from github. Like: 

gem 'sunspot_rails', :git => 'https://github.com/sunspot/sunspot.git'
group :development do
gem 'sunspot_solr', :git => 'https://github.com/sunspot/sunspot.git'
end

in your Gemfile. (Both the sample app and your app. I think.)

**Comment by bragboy** [comment]
A new gem has been released. You may do a bundle update to get the latest version instead of pointing to master directly.

**Comment by JasonBarnabe** [comment]
The latest releases didn't seem to do it for me with 5.3.1. I had to get sunspot, sunspot_solr, sunspot_rails from GitHub for a working version. What I did:

1. Update Gemfile to get the gems from GitHub
2. Blow away the local solr directory
3. Update to Java 8 (it complained `Unsupported major.minor version 51.0` when running `rake sunspot:solr:run`)
4. `rake sunspot:solr:run`, cancel when it finishes doing stuff.
5. Copy the generated files schema.xml and solrconfig.xml onto the server.
6. Restart the server.
